src/tree/LocalRootTreeItemBase.ts

```diff
@@ -108,7 +110,13 @@ export abstract class LocalRootTreeItemBase<TItem extends ILocalItem, TProperty
             this._currentItems = undefined;
             this._failedToConnect = true;
             context.telemetry.properties.failedToConnect = 'true';
-            return this.getDockerErrorTreeItems(context, error);
+            const dockerInstalled = await isDockerInstalled();
```

Comment:
We should change the order of the logic here, or else `docker -v` will be called four times (once for each of Containers, Images, Networks, Volumes). On my system that would take more than 30 seconds.

Reply:
In my machine it take half second to execute the command. I don't think it is worth optimizing this code considering that it gets exected only when it fails to connect. Also caching the dockerInstalled state will require additional logic on resetting it.

Reply:
It is _absolutely_ worth optimizing this code. The average execution time for any Docker command on Windows is about 2 seconds, and half second on Linux/Mac. You're gonna make people wait 8 seconds for nothing?

Reply:
docker -v is light weight compare to other docker commands that queries the images and containers. The half second that I measured is on windows machine. Even say if it takes average 2 seconds we are not serially waiting for all tree items. It gets loads in parallel, so all of them get the result at the same time (2 seconds) instead of 6+ seconds. Caching will add complexity and user will still wait for 2 seconds or whatever the time it takes to run 'docker -v' command. Are you saying 40% of the users fails to connect to docker?

Reply:
Also note that when the docker is not installed the command 'docker -v' takes 10 milliseconds.

Reply:
I don't think the caching code is actually difficult enough to justify not using it. @karolz-ms what do you think?

Reply:
I have one suggestion. In the base class, we could test `this.treePrefix === 'containers'` before doing an install check. For the rest of the views, it is maybe OK to show only the "Additional Troubleshooting" node, and not the "Docker not installed" / "Docker not running" warnings.

This way, we only do one `docker -v`, and avoid duplicating errors in each pane.

Reply:
We can use the `AsyncLazy` I'm working on, with a relatively short timeout of say 500ms. After all...the installation status is not going to change from not installed -> installed in <500ms, but most likely all four views will have attempted to refresh in this time.